theano/tensor/nnet/nnet.py

```diff
@@ -2049,3 +2050,121 @@ def relu(x, alpha=0):
         f1 = 0.5 * (1 + alpha)
         f2 = 0.5 * (1 - alpha)
         return f1 * x + f2 * abs(x)
+
+
+def h_softmax(x, batch_size, n_outputs, W1, b1, W2, b2,
+              n_classes, n_outputs_per_class, target=None):
```

Comment:
One last very minor modification. Now that they are no longer optional, it would make more sense to place ```n_classes``` and ```n_outputs_per_class``` after ```n_outputs```. It's actually the current order in the docstring.